card/mod/standard/set/all/links.rb

```diff
@@ -1,88 +1,66 @@
 RESOURCE_TYPE_REGEXP = /^([a-zA-Z][\-+\.a-zA-Z\d]*):/
 
-format :html do
-  def link_to text=nil, opts={}
-    opts[:href] ||= path opts.delete(:path)
-    text = raw(text || opts[:href])
-    interpret_data_opts_to_link_to opts
-    wrap_with :a, text, opts
-  end
-
-  def interpret_data_opts_to_link_to opts
-    %i[remote method].each do |key|
-      next unless (val = opts.delete key)
-      opts["data-#{key}"] = val
-    end
-  end
-end
-
-format :css do
-  def link_to _text=nil, opts={}
-    card_url path(opts.delete(:path))
-  end
-end
-
+# The #link_to methods support smart formatting of links in multiple formats.
 format do
+  # Creates a "link", the meaning of which depends upon the format.  In this base
+  # format, the link looks like [text][absolute path]
+  #
+  # @param text [String] optional string associated with link
+  # @param opts [Hash] optional Hash. In simple formats, :path is usually the only key
   def link_to text=nil, opts={}
-    path = path opts.delete(:path)
+    path = path((opts.delete(:path) || {}))
     if text && path != text
       "#{text}[#{path}]"
     else
       path
     end
   end
 
-  def link_to_resource resource, text=nil, opts={}
-    case (resource_type = resource_type resource)
-    when "external-link" then opts[:target] ||= "_blank"
-    when "internal-link" then resource = internal_url resource[1..-1]
-    end
-    add_class opts, resource_type
-    link_to text, opts.merge(path: resource)
-  end
-
-  def resource_type resource
-    case resource
-    when /^https?\:/          then "external-link"
-    when %r{^/}               then "internal-link"
-    when /^mailto\:/          then "email-link"
-    when RESOURCE_TYPE_REGEXP then Regexp.last_match(1) + "-link"
-    end
+  # link to a different view of the current card
+  # @param view [Symbol,String]
+  # @param text [String]
+  # @param opts [Hash]
+  def link_to_view view, text=nil, opts={}
```

Comment:
the main difference is that these methods now have their html/other format handling much more cleanly delineated.